modules/core/src/main/scala/lucuma/odb/api/model/ScienceRequirements.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats.Eq
+import cats.syntax.all._
+import cats.data.State
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import lucuma.core.optics.syntax.lens._
+import lucuma.odb.api.model.syntax.input._
+import clue.data.Input
+import monocle.Lens
+import monocle.Focus
+
+final case class ScienceRequirements(
+  mode:                     ScienceMode,
+  spectroscopyRequirements: SpectroscopyScienceRequirements
+)
+
+object ScienceRequirements {
+  val Default: ScienceRequirements = ScienceRequirements(ScienceMode.Spectroscopy, SpectroscopyScienceRequirements.Default)
+
+  val mode: Lens[ScienceRequirements, ScienceMode]                                         = Focus[ScienceRequirements](_.mode)
+  val spectroscopyRequirements: Lens[ScienceRequirements, SpectroscopyScienceRequirements] = Focus[ScienceRequirements](_.spectroscopyRequirements)
+
+  implicit val eqScienceRequirements: Eq[ScienceRequirements] =
+    Eq.by(x => (x.mode, x.spectroscopyRequirements))
+}
+
+object ScienceRequirementsModel {
+  final case class Create(
+    mode: ScienceMode,
+    spectroscopyRequirements: SpectroscopyScienceRequirementsModel.Input
+  ) {
+    val create: ValidatedInput[ScienceRequirements] =
+      spectroscopyRequirements.create.map { s =>
+        ScienceRequirements(mode, s)
+      }
+  }
+
+  object Create {
+    val Default: Create = Create(ScienceMode.Spectroscopy, SpectroscopyScienceRequirementsModel.Input.Default)
+
+    implicit val DecoderCreate: Decoder[Create] = deriveDecoder
+
+  }
+
+  final case class Edit(
+    mode:                     Input[ScienceMode]                                = Input.ignore,
+    spectroscopyRequirements: Input[SpectroscopyScienceRequirementsModel.Input] = Input.ignore
+  ) {
+
+    def editor: ValidatedInput[State[ScienceRequirements, Unit]] =
+      (mode.validateIsNotNull("mode"),
```

Comment:
I couldn't find how to make this optional. We may want to let the frontend edit without specifying the mode or the requirements.

Reply:
You want to make it possible to not specify the mode?  If so, in the model it should be an `Option` and you'd `validateNullable` here.

Reply:
mode can't really be an Option, I guess I'd have to pass it always. On the other hand the specific mode params can be optional.
I'm not sure why `validateNullable` takes a function as a param.

Reply:
If it is nullable (in that you can specify `null` and remove the optional field), and there is nothing to validate if it is specified, then you can use `toOptionOption` down in the state editor.  See `AsterismModel.Edit` and the `name` field.  It is nullable but there's nothing to validate.

Reply:
I think here for the `mode` you can't unset it, you can only change the value (?). In that case it makes sense to me that it is `validateIsNotNull`.